File: rei-do-chopp-site/src/app/finances/operation-histories/models/requests/operation-history-list.request.ts
import { PaginationRequest } from '../../../../shared/models/requests/pagination.request';
export class OperationHistoryListRequest {
  InitialDate: string;
  FinalDate: string;
  ProductsIds: Array<number>;

  constructor(params: Partial<{ InitialDate: Date, FinalDate: Date, ProductsIds: Array<number>}>) {
    this.InitialDate = params.InitialDate.toISOString() || null;
    this.FinalDate = params.FinalDate.toISOString() || null;
    this.ProductsIds = params.ProductsIds || [];
  }
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/OpertaionsHistories/Requests/OperationHistoryListRequest.cs
using System;

namespace ReiDoChopp.DataTransfer.OpertaionsHistories.Requests
{
    public class OperationHistoryListRequest
    {
        public DateTime? InitialDate { get; set; }
        public DateTime? FinalDate { get; set; }
        public int[] ProductsIds { get; set; }
    }
}
